### PR TITLE
ci: add Dependabot recursive go mod tidy support

### PR DESCRIPTION
**Pull request by zivkovicmilos**
## Description

This PR introduces a simple workflow that tidies go mod files on Dependabot PRs, and turns off the mod check for those PRs (to avoid overlapping workflows).

<details><summary>Contributors' checklist...</summary>

- [x] Added new tests, or not needed, or not feasible
- [x] Provided an example (e.g. screenshot) to aid review or the PR is self-explanatory
- [x] Updated the official documentation or not needed
- [x] No breaking changes were made, or a `BREAKING CHANGE: xxx` message was included in the description
- [ ] Added references to related issues and PRs
- [ ] Provided any useful hints for running manual tests
- [ ] Added new benchmarks to [generated graphs](https://gnoland.github.io/benchmarks), if any. More info [here](https://github.com/gnolang/gno/blob/master/.benchmarks/README.md).
</details>

### CODE REVIEW

File: .github/workflows/misc.yml [hunk 1]
```diff
@@ -58,8 +58,9 @@ jobs:
           else
             echo 'Succeeded.'
           fi
-  modtidy:
+  mod_tidy_check:
     runs-on: ubuntu-latest
+    if: ${{ github.actor != 'dependabot[bot]' }}
```
Comment (moul) [inline .github/workflows/misc.yml:63]: Why skipping for dependable?

File: .github/workflows/dependabot-tidy.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+          changes=$(git status --porcelain)
+          
+          if [ -n "$changes" ]; then
+            git config --local user.email "actions@github.com"
+            git config --local user.name "GitHub Actions"
+            git add -A
+            git commit -m "Run 'go mod tidy' via GitHub Actions" && git push
+          else
+            echo "No changes to Go mods. Skipping commit"
+          fi
```
[the remark this answers is not in the document]
Comment (moul) [reply .github/workflows/dependabot-tidy.yml:49]: They were, but wasn’t obvious

**Review by moul — APPROVED** [review]

**Comment by ajnavarro** [comment]
Maybe that can be done using [dependabot groups](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#groups)

**Comment by zivkovicmilos** [comment]
> Maybe that can be done using [dependabot groups](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#groups)

I love this 💯 

I'm gonna push a PR soon for the gno repo to add these groups, but also for our other repos (JS clients especially, this will save my sanity)

File: .github/workflows/dependabot-tidy.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (zivkovicmilos) [reply .github/workflows/dependabot-tidy.yml:50]: Yeah, it was bundled with another command; I've applied @moul's suggestion:
https://github.com/gnolang/gno/blob/65ad24640edb3ef9b03ac6e106a113f53f93664a/.github/workflows/dependabot-tidy.yml#L47

[the remark this answers is not in the document]
Comment (zivkovicmilos) [reply .github/workflows/dependabot-tidy.yml]: @thehowl good catch 💯 

I've swapped this commit step to use the action:
https://github.com/gnolang/gno/pull/1412/commits/345a5cf8021eded12d2c5fb11549a196cf46ec7d

**Review by thehowl — APPROVED** [review]

**Pull request merged**